pkg/controllers/management/auth/manager.go

```diff
@@ -755,11 +770,24 @@ func checkGroup(apiGroup string, rule v1.PolicyRule) bool {
 	return false
 }
 
+// reconcileManagementPlaneRole ensures that the management plane role exists and has the right rules.
+// If not, update or create it as needed.
 func (m *manager) reconcileManagementPlaneRole(namespace string, resourceToVerbs map[string]map[string]string, rt *v3.RoleTemplate) error {
-	roleCli := m.mgmt.RBAC.Roles(namespace)
+	// If the namespace isn't available or terminating don't proceed
+	ns, err := m.nsLister.Get("", namespace)
+	if err != nil {
+		return errors.Wrapf(err, "couldn't get namespace %v", namespace)
+	}
+	if ns.Status.Phase == v12.NamespaceTerminating {
+		logrus.Warnf("[%v] Namespace %v is terminating, not creating role", m.controller, namespace)
+		return nil
+	}
+
```

Comment:
This warning will appear for every roles inherited by the `RoleTemplate`. I think it would be better instead to break out of the loop if once we find that the namespace is terminating. What do you think?

OR

Add the name of the role that we couldn't create, to differentiate the logs.